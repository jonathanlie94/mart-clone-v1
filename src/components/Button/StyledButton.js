import styled from 'styled-components';

export default styled.button`
  height: auto;
  text-align: center;
  outline-width: 0;
  border-color: transparent;
  border-radius: 2px;
  color: ${props => props.color};
  background-color: ${props => props.backgroundColor};
  &:focus:not([disabled]),
  :hover:not([disabled]) {
    opacity: .625;
  }
  &:disabled {
    opacity: .25;
  }
  transition: .1s ease;
`;
