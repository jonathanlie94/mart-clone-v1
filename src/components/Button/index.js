import React from 'react';
import PropTypes from 'prop-types';
import theme from 'ui/theme';
import StyledButton from './StyledButton';

const Button = props => (
  <StyledButton
    className={props.className}
    color={props.color}
    disabled={props.disabled}
    backgroundColor={props.backgroundColor}
    onClick={props.disabled ? null : props.onClick || null}
  >
    {props.children}
  </StyledButton>
);

Button.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: '',
  children: false,
  color: theme.colors.white,
  backgroundColor: theme.colors.primary,
  disabled: false,
  primary: false,
};

export default Button;
