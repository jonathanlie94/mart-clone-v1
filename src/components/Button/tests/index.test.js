import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import Button from '../';

test('should render children when passed in', () => {
  const renderedComponent = shallow(
    <Button onClick={() => {}}>
      <div className="unique-class-name" />
    </Button>
  );
  expect(
    renderedComponent.contains(<div className="unique-class-name" />)
  ).toEqual(true);
});

test('should render a button', () => {
  const button = shallow(<Button onClick={() => {}} />).dive();
  expect(button.type()).toEqual('button');
});

test('should match previous snapshot', () => {
  const button = shallow(<Button onClick={() => {}} />).dive();
  const tree = toJSON(button);
  expect(tree).toMatchStyledComponentsSnapshot();
});

test('calls onClick props when clicked', () => {
  const onClickSpy = jest.fn();
  const renderedComponent = mount(<Button onClick={onClickSpy} />);
  renderedComponent.find('button').simulate('click');
  expect(onClickSpy).toHaveBeenCalled();
});
