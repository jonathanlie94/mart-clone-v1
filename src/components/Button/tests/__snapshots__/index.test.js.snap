// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
.kFNwvQ {
  height: auto;
  text-align: center;
  outline-width: 0;
  border-color: transparent;
  border-radius: 2px;
  color: #FFF;
  background-color: #FFC70E;
  -webkit-transition: .1s ease;
  transition: .1s ease;
}

.kFNwvQ:focus:not([disabled]),
.kFNwvQ:hover:not([disabled]) {
  opacity: .625;
}

.kFNwvQ:disabled {
  opacity: .25;
}

<button
  className="sc-bdVaJa kFNwvQ"
  color="#FFF"
  disabled={false}
  onClick={[Function]}
/>
`;
